cpp/src/arrow/CMakeLists.txt

```diff
@@ -149,6 +143,17 @@ set(ARROW_SRCS
     util/utf8.cc
     vendored/datetime/tz.cpp)
 
+if(ARROW_JSON)
```

Comment:
I think the `add_subdirectory(json)` line also needs to be moved here

Reply:
So it does!  Thanks for the good spot.